docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,262 @@
+to allow limited access to image registries.
+
+### Prep 3: Images
+
+All workloads in a Kubernetes cluster are running as an OCI image.
+Kubernetes needs to be able to fetch these images and load them
+into the container runtime, in order to run workloads.
+For a Canonical Kubernetes deployment, you will need to fetch the images
+used by its features (network, dns, etc) as well as any images that are
+needed to run your workloads.
+
+The following options are presented in the order of
+increasing complexity of implementation.
+You may also find it helpful to combine these options for your scenario.
+
+If you already have a running cluster,
+you can list the images in use by running:
+
+```bash
+sudo k8s list-images
+```
+
+Please remember to keep track of the images used by your workloads as well.
+
+#### Images Option A: via an HTTP proxy
+
+In many cases, the nodes of the airgap deployment may not have direct access to
+upstream registries, but can reach them through the
+[use of an HTTP proxy][proxy].
+
+The configuration of the proxy is out of the scope of this documentation.
+
+#### Images Option B: private registry mirror
+
+In case regulations and/or network constraints do not allow the cluster nodes
+to access any upstream image registry,
+it is typical to deploy a private registry mirror.
+This is an image registry service that contains all the required OCI Images
+(e.g. [registry](https://distribution.github.io/distribution/),
+[Harbor](https://goharbor.io/) or any other OCI registry) and
+is reachable from all cluster nodes.
+
+This requires three steps:
+
+1. Deploy and secure the registry service. This is out of the scope for this
+   document, please follow the instructions for the registry
+   that you want to deploy.
+2. Using [regsync][regsync], load all images from the upstream source and
+   push to your registry mirror.
+3. Configure the Canonical Kubernetes container runtime (`containerd`) to load 
+   images from
+   the private registry mirror instead of the upstream source. This will be
+   described in the installation section.
+
+In order to load images into the private registry, you need a machine with
+access to both the upstream registry (e.g. `docker.io`) and the internal one. 
+
+##### Load images with regsync
+
+We recommend using [regsync][regsync] to copy images
+from the upstream registry to your private registry.
+
+For the images used in the k8s-snap we currently sync upstream images
+to `ghcr.io/canonical`.
+
+Since you will need to do something similar you
+will find it helpful to look at the [upstream-images.yaml][upstream-imgs] file
+as well as the [sync-images][sync-images] script.
+
+In [upstream-images.yaml][upstream-imgs] you will have to
+change the sync target from `ghcr.io/canonical` to your private registry mirror.
+
+After you have updated the yaml file, you can run the [sync-images][sync-images]
+script:
+
+```bash
+USERNAME="$username" PASSWORD="$password" ./sync-images.sh
+```
```

Comment:
This is not sufficient, as it does not cover our rocks. Instead, we should point people to use `k8s list-images` to get the full list of images (or look inside the download snap and check for the `images.txt` file).

Then, we should provide an example file, similar to `upstream-images.yaml`, that syncs images _from_ `ghcr.io` to the target mirror (which can also be configured through env. This can look like this:

```yaml
sync:
  - source: ghcr.io/canonical/k8s-snap/pause:3.10
    target: '{{ env "MIRROR" }}/canonical/k8s-snap/pause:3.10'
    type: image
  # ....
```